### DISCUSSION TITLE
meaning of v42_irrig_eff

### DISCUSSION THREAD

**Discussion by jansteinhauser** [magpiemodel/magpie#296] [Q&A] [answered]
Hello everyone,

Very excited to work on/with MAgPIE but still getting the hang of it. So, I may have a few basic questions here and there. The first one is a sanity check;

In [Water Demand](https://rse.pik-potsdam.de/doc/magpie/4.3/42_water_demand.htm), the variable **v42_irrig_eff** is described as "correspond[ing] to the amount of water that is used inefficiently in the irrigation process". 

However, the way I understand the demand equation, the variable describes the share of the total (crop) water demand that actually reaches the crops, while, for example, in the global static scenario with, by default, `v42_irrig_eff <- 0.66`, 34% of water is used inefficiently and lost without effect. Meaning, at 66% efficiency crops would have an actual demand of  ~151% of their base demand. 

This, in my understanding, is the exact opposite of the variable description quoted above. So, my question is: Is this interpretation correct or do I misunderstand the variable description, equation, or both?

Thank you and all the best,
Jan

**Edit** for clarity, as I was imprecise when calling the global static efficiency the default (meaning the default efficiency in that scenario). It was just intended as a calculation example.

**Comment by abhimishr** [comment]
Hi Jan,

The [default realization](https://github.com/jansteinhauser/magpie/blob/455924c544dd6d14c827c6c23b5f0068f305f2d3/config/default.cfg#L624) of water demand module (`42_water_demand`) seems to be `agr_sector_aug13`. 

There, `v42_irrig_eff` is always fixed to a certain value depending on `s42_irrig_eff_scenario` ([see here](https://github.com/jansteinhauser/magpie/blob/455924c544dd6d14c827c6c23b5f0068f305f2d3/modules/42_water_demand/agr_sector_aug13/presolve.gms#L58)). 

Default `s42_irrig_eff_scenario` is set to `2` (from [here ](https://github.com/jansteinhauser/magpie/blob/455924c544dd6d14c827c6c23b5f0068f305f2d3/config/default.cfg#L646) and [here](https://github.com/jansteinhauser/magpie/blob/455924c544dd6d14c827c6c23b5f0068f305f2d3/modules/42_water_demand/agr_sector_aug13/input.gms#L12)) and is described as "[regional static values from gdp regression](https://github.com/jansteinhauser/magpie/blob/455924c544dd6d14c827c6c23b5f0068f305f2d3/config/default.cfg#L644)".  

In case you have a MAgPIE run available. You can plug in the `fulldata.gdx` file and check the values using following code snippet in an R environment:

```
library(magpie4)
gdx <- "path/to/fulldata.gdx"
irrig_eff <- collapseNames(readGDX(gdx,"ov42_irrig_eff")[,,"level"])
irrig_eff
```

As `s42_irrig_eff_scenario` is set to `2`, every MAgPIE cluster would get a specific value based on the regression calculation from [here](https://github.com/jansteinhauser/magpie/blob/455924c544dd6d14c827c6c23b5f0068f305f2d3/modules/42_water_demand/agr_sector_aug13/presolve.gms#L59). A global value of `0.66` is only used if `s42_irrig_eff_scenario` is set to `1`.

From the documentation link, it looks like this was generated from in-code documentation which was last updated [quite a long time ago](https://github.com/jansteinhauser/magpie/blame/17185904447c64a90228b982a4506b83caaccb5b/modules/42_water_demand/agr_sector_aug13/equations.gms). Meanwhile, I am tagging our irrigation experts @FelicitasBeier and @vartika271987 here as they would be the best ones to clarify it further.

Hope this helps!

**Reply by jansteinhauser** [reply]
Hi Abhijeet,

Thank you for your explanation. I've edited my initial question as I realized my wording was quite imprecise. Your explanation was definitely helpful to drive home the differences in the scenario choices but my question was more basic than that. Basically, I just wanted to ensure that I understood the meaning of the variable correctly. 
To me, based on the equation, it's the share of water that reaches the crops. However, the text reads to me as if it was the share of water that does not. 

Thanks again!

**Reply by FelicitasBeier** [reply]
v42_irrig_eff(j) indeed refers to Irrigation efficiency (as defined in the declarations). It is the share of water that is actually used by the crop(s).
To calculate water demand for agriculture we calculate water use of crops (area*water_requirements*irrigation_efficiency) + water use of livestock (livestock_production * water_requirements_of_livestock) (as you can see in the respective equation).
Losses of water that occur through transport from source to field are not consumed by the crops and therefore should not be accounted as agricultural water use. 

I will have a look at the wording of the in-code comments again. It seems that this might have been phrased misleadingly. The variable `v42_irrig_eff` indeed captures inefficient water use (which I assume the author wanted to say here), but it does not correspond to it, in my view. 
You can also have a look at the realization.gms of the respective modules for further explanations. 

Hope this was helpful.

**Reply by jansteinhauser** [reply]
Thank you for clarifying, seems I got it right then. 
I can't mark your response as answer. Could you repost your text as direct answer to my initial question please?

Thanks again!

**Comment by FelicitasBeier** [comment, marked as answer]
v42_irrig_eff(j) indeed refers to Irrigation efficiency (as defined in the declarations). It is the share of water that is actually used by the crop(s).
To calculate water demand for agriculture we calculate water use of crops (areawater_requirementsirrigation_efficiency) + water use of livestock (livestock_production * water_requirements_of_livestock) (as you can see in the respective equation).
Losses of water that occur through transport from source to field are not consumed by the crops and therefore should not be accounted as agricultural water use.

I will have a look at the wording of the in-code comments again. It seems that this might have been phrased misleadingly. The variable v42_irrig_eff indeed captures inefficient water use (which I assume the author wanted to say here), but it does not correspond to it, in my view.
You can also have a look at the realization.gms of the respective modules for further explanations.

Hope this was helpful.
